### PR TITLE
chore: Enable restart and reconnect tests in CI under a different Tag

### LINKED ISSUE #10581: Add Restart and Reconnect HapiTests to CI
Add `MixedOpsRestartTest` and `MixedOpsNodeDeathReconnectTest` to CI checks

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
WIP

Fixes #10581 

- Since restart and reconnect tests can't be run along with other tests, added them under different tag to be run in CI.
- Opening as draft PR to see if the CI run succeeds

### CODE REVIEW

File: .github/workflows/node-zxc-compile-application-code.yaml [hunk 1]
```diff
@@ -386,6 +396,56 @@ jobs:
...
+          json_thousands_separator: ','
+          junit_files: "**/build/test-results/hapiTestNDReconnect/TEST-*.xml"
+
+      - name: Publish HAPI Test (Node Death Reconnect) Network Logs
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        if: ${{ inputs.enable-hapi-tests-nd-reconnect && inputs.enable-network-log-capture && !cancelled() }}
+        with:
+          name: HAPI Test (Time Consuming) Network Logs
+          path: |
+            hedera-node/test-clients/build/hapi-test/**/output/**
```
Comment (nathanklick) [inline .github/workflows/node-zxc-compile-application-code.yaml:447]:
```suggestion
          name: HAPI Test (Node Death Reconnect) Network Logs
          path: |
            hedera-node/test-clients/build/hapi-test/**/output/**
            hedera-node/test-clients/build/hapi-test/*.log
```

File: .github/workflows/node-zxc-compile-application-code.yaml [hunk 2]
```diff
@@ -386,6 +396,56 @@ jobs:
...
+          json_thousands_separator: ','
+          junit_files: "**/build/test-results/hapiTestRestart/TEST-*.xml"
+
+      - name: Publish HAPI Test (Restart) Network Logs
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        if: ${{ inputs.enable-hapi-tests-restart && inputs.enable-network-log-capture && !cancelled() }}
+        with:
+          name: HAPI Test (Restart) Network Logs
+          path: |
+            hedera-node/test-clients/build/hapi-test/**/output/**
```
Comment (nathanklick) [inline .github/workflows/node-zxc-compile-application-code.yaml:422]:
```suggestion
            hedera-node/test-clients/build/hapi-test/**/output/**
            hedera-node/test-clients/build/hapi-test/*.log
```

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

File: .github/workflows/node-flow-pull-request-checks.yaml [hunk 3]
```diff
@@ -228,6 +228,44 @@ jobs:
       gradle-cache-password: ${{ secrets.GRADLE_CACHE_PASSWORD }}
       sonar-token: ${{ secrets.SONAR_TOKEN }}
 
+  hapi-tests-restart:
+    name: HAPI Tests (Restart)
+    uses: ./.github/workflows/node-zxc-compile-application-code.yaml
+    needs:
+      - build
```
Comment (JeffreyDallas) [inline .github/workflows/node-flow-pull-request-checks.yaml:235]: Please change to:
    needs:
      - dependency-check
      - spotless

File: .github/workflows/node-flow-pull-request-checks.yaml [hunk 4]
```diff
@@ -228,6 +228,44 @@ jobs:
...
+      access-token: ${{ secrets.GITHUB_TOKEN }}
+      gradle-cache-username: ${{ secrets.GRADLE_CACHE_USERNAME }}
+      gradle-cache-password: ${{ secrets.GRADLE_CACHE_PASSWORD }}
+      sonar-token: ${{ secrets.SONAR_TOKEN }}
+
+  hapi-tests-nd-reconnect:
+    name: HAPI Tests (Node Death Reconnect)
+    uses: ./.github/workflows/node-zxc-compile-application-code.yaml
+    needs:
+      - build
```
Comment (JeffreyDallas) [inline .github/workflows/node-flow-pull-request-checks.yaml:254]: same here ...

**Review by JeffreyDallas — APPROVED** [review]

File: .github/workflows/node-flow-pull-request-checks.yaml (hunk 4 above)
Comment (Neeharika-Sompalli) [reply .github/workflows/node-flow-pull-request-checks.yaml]: Fixed Jeffrey. Thanks

File: .github/workflows/node-flow-pull-request-checks.yaml (hunk 3 above)
Comment (Neeharika-Sompalli) [reply .github/workflows/node-flow-pull-request-checks.yaml]: Fixed Jeffrey. Thanks

**Review by JeffreyDallas — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

**Review by nathanklick — APPROVED** [review]

**Pull request merged**